rest/config.go

```diff
@@ -1221,11 +1214,11 @@ func ServerMain() {
 	}
 
 	// Validation
-	var errorMsgs = make([]error, 0)
-	errorMsgs = append(errorMsgs, config.validate()...)
-	errorMsgs = append(errorMsgs, config.setupAndValidateDatabases()...)
-	if len(errorMsgs) > 0 {
-		for _, err := range errorMsgs {
+	var errorMsgs *multierror.Error
+	errorMsgs = multierror.Append(errorMsgs, config.validate())
+	errorMsgs = multierror.Append(errorMsgs, config.setupAndValidateDatabases())
+	if errorMsgs != nil {
```

Comment:
This is preventing SG from starting up at all, which is what was causing the LiteCore tests to fail.

```
14:32 $ go1.13.4 build && ./sync_gateway sg_config.json
2020-10-01T14:32:55.503+01:00 ==== /() CE ====
2020-10-01T14:32:55.504+01:00 [INF] Logging: Console to stderr
2020-10-01T14:32:55.504+01:00 [INF] Logging: Files to /tmp/sglogs1
2020-10-01T14:32:55.504+01:00 [ERR] Error(s) during config validation -- rest.ServerMain() at config.go:1224
```

`errorMsgs` here isn't nil, but might still have zero errors after both appends have run.

This is the type of situation where the `ErrorOrNil()` util function should be used.

Need to go through all usages of `!= nil` and determine whether required elsewhere.

Reply:
Yeah, making use of ErrorOrNil here and in all other places is ideal.